core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala

```diff
@@ -43,7 +43,7 @@ class IssueTestSuite extends BaseTiSparkTest {
     }
     spark.sqlContext.setConf(TiConfigConst.USE_INDEX_SCAN_FIRST, "true")
     tidbStmt.execute("""
-        |SET tidb_enable_clustered_index = 1;
+        |SET GLOBAL tidb_enable_clustered_index = 1;
```

Comment:
It seems that the global variable cannot take effect immediately after you set it in one session.

> Similar to MySQL, changes to GLOBAL variables do not apply to either existing connections, or the local connection. Only new sessions reflect the changes to the value.

https://docs.pingcap.com/tidb/stable/sql-statement-set-variable#set-globalsession-variable

Reply:
got thx